src/kOS/Module/kOSProcessor.cs

```diff
@@ -30,6 +30,15 @@ public class kOSProcessor : PartModule, IProcessor, IPartCostModifier, IPartMass
 
         public Harddisk HardDisk { get; private set; }
 
+        public string Tag
+        {
+            get
+            {
+                KOSNameTag tag = part.Modules.OfType<KOSNameTag>().FirstOrDefault();
+                return tag != null ? tag.nameTag : null;
```

Comment:
Since kOS doesn't give the user any way to interact with null values, this should return an empty string.  There is no guarantee that the tag will be present on a given part, because the MM script was intentionally written so that the user could remove the name tag from any part using the :final keyword.

From the PartValue class: `return tagModule == null ? string.Empty : tagModule.nameTag;`